build/git_revision.go

```diff
@@ -169,45 +174,71 @@ func (gr *GitRevision) Build(ctx context.Context) (string, error) {
 		installDir = tmpDir
 		gr.pathsToRemove = append(gr.pathsToRemove, installDir)
 	}
+	gr.log().Printf("install dir is %q", installDir)
 
 	// copy license file on best effort basis
-	dstLicensePath := filepath.Join(installDir, "LICENSE.txt")
-	srcLicensePath := filepath.Join(repoDir, "LICENSE.txt")
-	altSrcLicensePath := filepath.Join(repoDir, "LICENSE")
-	if _, err := os.Stat(srcLicensePath); err == nil {
-		err = gr.copyLicenseFile(srcLicensePath, dstLicensePath)
-		if err != nil {
-			return "", err
-		}
-	} else if _, err := os.Stat(altSrcLicensePath); err == nil {
-		err = gr.copyLicenseFile(altSrcLicensePath, dstLicensePath)
-		if err != nil {
-			return "", err
-		}
+	// default to installDir if LicenseDir is not set
+	licenseDir := gr.LicenseDir
+	if licenseDir == "" {
+		licenseDir = installDir
+	}
+	gr.log().Printf("Attempting to copy license file to %q", licenseDir)
+	if err := gr.copyLicenseIfExists(repoDir, licenseDir); err != nil {
+		return "", err
 	}
 
 	gr.log().Printf("building %s (timeout: %s)", gr.Product.Name, buildTimeout)
 	defer gr.log().Printf("building of %s finished", gr.Product.Name)
 	return bi.Build.Build(buildCtx, repoDir, installDir, gr.Product.BinaryName())
 }
 
+func (gr *GitRevision) copyLicenseIfExists(repoDir string, dstDir string) error {
+	licenseFiles := []string{"LICENSE.txt", "LICENSE"}
+
+	for _, file := range licenseFiles {
+		srcPath := filepath.Join(repoDir, file)
+		if _, err := os.Stat(srcPath); err == nil {
+			gr.log().Printf("Found license file at %q", srcPath)
+			dstPath := filepath.Join(dstDir, file)
+			if err := gr.copyLicenseFile(srcPath, dstPath); err != nil {
+				return fmt.Errorf("failed to copy license file from %q to %q: %w", srcPath, dstPath, err)
+			}
+		}
+	}
+
+	return nil
+}
+
 func (gr *GitRevision) copyLicenseFile(srcPath, dstPath string) error {
+	gr.log().Printf("Copying license file from %q to %q", srcPath, dstPath)
 	src, err := os.Open(srcPath)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to open license file at %q: %w", srcPath, err)
 	}
 	defer src.Close()
+
+	// Create the directory if it does not exist
+	dir := filepath.Dir(dstPath)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		gr.log().Printf("Directory %q does not exist, creating it", dir)
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			return fmt.Errorf("failed to create directory %q: %w", dir, err)
+		}
+	}
```

Comment:
It's true we automatically create _temporary_ installation directory (if `InstallDir` is empty) but we don't do that for static paths as we expect the consumer to provide a path that is already managed by them (i.e. has been created).

I think we should follow the same logic here to avoid unnecessary edge cases.